core/src/main/java/org/elasticsearch/indices/store/IndicesStore.java

```diff
@@ -115,11 +123,29 @@ public void clusterChanged(ClusterChangedEvent event) {
             return;
         }
 
-        for (IndexRoutingTable indexRoutingTable : event.state().routingTable()) {
+        RoutingTable routingTable = event.state().routingTable();
+
+        // remove entries from cache that don't exist in the routing table anymore
```

Comment:
can we add a note that these are either closed indices (shouldn't be deleted) or deleted indices that are dealt with somewhere else (link to where)?